### DISCUSSION TITLE
Does it make sense to make ekg-remove-unused-tags interactive?

### DISCUSSION THREAD

**Discussion by jayrajput** [ahyatt/ekg#54] [Q&A]
I imported my `org-roam` notes to ekg and ended up with duplicate notes with separate tags differing by underscores. So I am deleting a lot of duplicate notes and tags, so reaching out to to `ekg-clean-db`, but then discovered `ekg-remove-unused-tags`. Since `ekg-remove-unused-tags` is not interactive, I keep forgetting the name and have to look at the documentation to find it again. I am also working in parallel to find such duplicates notes/tags programmatically and remove such duplicate notes.

Does it make sense to make `ekg-remove-unused-tags` as interactive, so that it can be invoked using `M-x`?

**Comment by ahyatt** [comment]
I think that seems reasonable.  My only question is whether it is necessary to do the same forced back up that `ekg-clean-db` does before just removing the tags.  I think not - however it would also be useful to output the set of tags that were cleaned up, just so the user knows.  I can look into doing this soon, although first I would like to release the current set of changes.

**Reply by jayrajput** [reply]
Shall such deleted tags end up in trash as well? My use cases is to not make them not visible when searching the ekg notes.

**Reply by ahyatt** [reply]
Right now a tag can't end up in the trash, only notes that contain tags.

**Reply by jayrajput** [reply]
Ok. So we can just remove the tag. A backup seems to be overkill. I do appreciate the paranoia though in exposing this function to the end user though. So if you decide to not make it interactive, that seems to be fair.

For my workflow, where I am backing up the notes in git repo by exporting them using logseq export, I have a plan B in case things go bad (although there is no reason I can think now why things can go bad). But a standard user who is relying on the triples backup, there is no other backup.
